src/par_iter/collect/mod.rs

```diff
@@ -1,16 +1,35 @@
-use super::ExactParallelIterator;
+use super::{ParallelIterator, ExactParallelIterator};
 use std::isize;
 use std::sync::atomic::{AtomicUsize, Ordering};
 
 mod consumer;
 use self::consumer::CollectConsumer;
 
+/// Collects the results of the exact iterator into the specified vector.
 pub fn collect_into<PAR_ITER,T>(mut pi: PAR_ITER, v: &mut Vec<T>)
     where PAR_ITER: ExactParallelIterator<Item=T>,
-          PAR_ITER: ExactParallelIterator,
           T: Send,
 {
     let len = pi.len();
+    special_collect_into(pi, len, v);
```

Comment:
Instead of adding `special_collect_into` as a method, can we make it a free fn in the `collect` module and call *that*?

(In general, I'd prefer to have fewer `doc(hidden)` methods, not more.)